Update README (#76)
onmax/nimiq-validator-trust-score: README.md

  <a id="README" href="#README" href="https://github.com/nimiq/core-rs-albatross/blob/albatross/README.md">
    <img src="https://raw.githubusercontent.com/nimiq/developer-center/refs/heads/main/assets/images/logos/validators-API.svg" alt="Nimiq PoS Albatross Repository" width="600" />
  </a>
</br>
</br>

[![Sync Mainnet](https://github.com/nimiq/validators-api/actions/workflows/sync-mainnet.yml/badge.svg)](https://github.com/nimiq/validators-api/actions/workflows/sync-mainnet.yml) [![Sync Testnet](https://github.com/nimiq/validators-api/actions/workflows/sync-testnet.yml/badge.svg)](https://github.com/nimiq/validators-api/actions/workflows/sync-testnet.yml)

> The Nimiq Validators API enables staking pools, based on single validators, to integrate with the Nimiq Wallet. This helps stakers make informed decisions when choosing where to stake their funds.

Reviewer: Mmm, that is one of the main features it provides, but not the only one.

The validators API is also there to support other applications, such as nimiq.watch, so that those applications can get the data from a single source of truth.

This API is being used in multiple places already

**Validators and Staking Pools**:
A validator can operate as a staking pool, allowing multiple users to stake. Pools must provide detailed information such as fees, payout schedules, and contact details to ensure trust and transparency.

> Stakers can evaluate a validator’s reliability using the Validator Trustscore (VTS) and review staking pool details like payout schedules to select the best option.

Reviewer 2: Do we go for Validator Trustscore or just Validator Score?

Reviewer: Trustscore

> Stakers can evaluate a validator’s reliability using the Validator Trust Score (VTS) and review staking pool details like payout schedules to select the best option.

Reviewer: ```suggestion
Stakers can evaluate a validator’s reliability using the Validator Trustscore (VTS) and review staking pool details like payout schedules to select the best option.
```

Reviewer: Maybe I am wrong, but we are going with the `Trustscore` together,  right?

## Add your Validator Information

> If you operate a staking pool and want to be displayed in the Nimiq Wallet, follow these steps:

Reviewer: Same as before, maybe something like:

```suggestion
If you operate a staking pool and want your information to be publicly available, such as in the Nimiq Wallet, follow these steps:
```

1. [Fork this repository](https://github.com/nimiq/validators-api/fork).
2. Add your Validator File:
   - Create a new JSON file in the `public/validators/main-albatross` directory.

>    - Use the provided example template in the directory to structure your data.

Reviewer: ```suggestion
   - Use the provided [example template](./public/validators/.example.json) in the directory to structure your data.
```

> 3. Review the [Description Guidelines](#recommendations-for-your-validator-information).

Reviewer: ```suggestion
3. Review the [Description Guidelines](#recommendations-for-your-validator-description).
```

4. Learn about the [JSON Schema](#validator-json-schema).

> 5. Submit a PR to this repository. A Nimiq team member will review your submission.

Reviewer: At the moment this process is quite opaque and slow (up to 3 days).

Can we add something about this? We need to be more transparent:

- Why it takes so much?
- What are we verifying exactly?

1. We need to verify the website contents
2. We need to verify that the fee stated by the validator is what actually is doing (not sure how this is being tested, ask Micha)...

Not sure if this is the place

Author: I talked to Micha and Richy. The PR will be reviewed in up to 3 days; fee verification is an ongoing process. In this README, I will only mention it in the PR verification. The fee verification time will be updated in the pool guidelines

> 6. Once the PR is submitted, check that the [API endpoint](https://validators-api-mainnet.nuxt.dev/api/v1) returns your information. This process may take a few minutes.

Reviewer: ```suggestion
6. Once the PR is submitted, check that the [API endpoint](https://validators-api-mainnet.nuxt.dev/api/v1) returns your information. This process may take a few days.
```

Sadly, this process is not automatic at the moment... we need to run it manually. This will be different in the near future

### Recommendations for your Validator Description

#### Description

This information will be displayed in the wallet to help stakers decide which pool they want to stake in. Providing clear and concise information helps stakers make informed decisions. For complete guidelines and suggestions, check our [Staking Pools Handbook](https://forum.nimiq.community/t/staking-pools-handbook/2169).

- **Length**: Aim for 1-2 sentences (20-40 words).
- **Clarity**: Highlight unique aspects of your pool, such as reliability, transparency or low fees.
- **Strengths**: Mention features like high uptime, low fees, strong community support or eco-friendly practices.
- **Tone**: Keep a friendly and professional tone.

**Good Example**: "Our pool offers 99.9% uptime, low fees, and strong security, ensuring a seamless staking experience. Join us for a reliable, transparent, and staker-focused service."
**Bad Example**: "Stake with us for unmatched rewards and the highest profits! Don’t miss your chance to earn big — this is the ultimate opportunity for stakers!"

### Validator JSON schema

> Use the following schema to create your validator information file. You can start by copying this [example JSON template](./public/validators/main-albatross/.example.json). When you add your validator information, you'll need to include specific keys in your JSON file. Below is an explanation of each key and its possible values.

Reviewer: ```suggestion
Use the following schema to create your validator information file. You can start by copying this [example JSON template](./public/validators/.example.json). When you add your validator information, you'll need to include specific keys in your JSON file. Below is an explanation of each key and its possible values.
```

**Required Keys**:

- `name`: The name of your validator. If you don’t provide a name, your validator address will be used by default.
- `description`: A short description of your validator. Use this to highlight what makes your validator unique or appealing to stakers.
- `address`: The address of the validator in this format `NQXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX`.

> - `fee`: A percentage fee between 0 and 1 (0.05 represents a 5% fee).

Reviewer: Not sure if this is clear enough 🙈 .

Maybe overthinking, but something like this might be clearer?

```md
- Fee information. You need to add ONE of the following:
   - For static fees, use `fee`. A percentage fee between 0 and 1 (0.05 represents a 5% fee). 
   - For dynamic fees, you need to add three keys:
     - `feeLowest`: The minimum possible fee.
     - `feeHighest`: The maximum possible fee.
     - `feeDescription`: Outline the conditions for various fees.
```

  - If using dynamic fees:
    - `feeLowest`: The minimum possible fee.
    - `feeHighest`: The maximum possible fee.
    - `feeDescription`: Outline the conditions for various fees.
- `payoutType`: The method used to payout the rewards. Possible values are:
  - `restake`: Rewards are automatically restaked.
  - `direct`: Rewards are paid directly into the staker's wallet and are not automatically restaked. Requires:
    - `payoutAddress`: Provide address you will payout from.

>     - `payoutSchedule`: Specifiy the frequency of payouts using the [cron job format](https://cloud.google.com/scheduler/docs/configuring/cron-job-schedules). Example: `0 */6 * * *` for payouts every 6 hours.

Reviewer: Maybe add also https://crontab.guru/ which is more beginner friendly?

pnpm dev
```

**Notes on Development**

- The project uses Nuxt Hub, which runs on Cloudflare Pages with a Cloudflare D1 database.

> - The database is a SQLite instance, shared remotely in both development and production environments. To change this behavior, modify the configuration in nuxt.config.ts.

Reviewer 2: This is constantly changing because NuxtHub is still in beta, so I would add a message saying that this may change and redirect to the NuxtHub docs.
For example, the database is not shared anymore, they have a [preview mode](https://hub.nuxt.com/docs/getting-started/remote-storage#production-vs-preview) that is the development db and the production one

Reviewer: Even remove this section? Not really necessary
